hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/S3AInputStream.java

```diff
@@ -561,6 +566,12 @@ public synchronized void close() throws IOException {
       try {
         // close or abort the stream; blocking
         awaitFuture(closeStream("close() operation", false, true));
+        // Collect ThreadLevel IOStats
+        if (ioStatisticsContext != null) {
+          ioStatisticsContext.getThreadIOStatistics().aggregate(streamStatistics.getIOStatistics());
```

Comment:
ioStatisticsContext.getThreadIOStatistics() needs to be cached in the constructor, so that even when closed *in a different thread* the original stat context is updated. this also guarantees that the reference doesn't get cleaned up while a stream is open